src/quicr_client.cpp

```diff
@@ -1,665 +1,113 @@
-    log_msg << "Transport failed to connect to server, got: "
-            << int(transport->status());
-    logger.log(qtransport::LogLevel::fatal, log_msg.str());
-    throw std::runtime_error(log_msg.str());
-  }
-
-
-  transport_stream_id = transport->createStream(transport_context_id, false);
-}
-
 ///
 /// QuicRClient
 ///
 
 QuicRClient::QuicRClient(RelayInfo& relay_info,
                          qtransport::TransportConfig tconfig,
                          qtransport::LogHandler& logger)
-  : log_handler(logger)
 {
-  log_handler.log(qtransport::LogLevel::info, "Initialize QuicRClient");
-
-  if (relay_info.proto == RelayInfo::Protocol::UDP) {
-    // For plain UDP, pacing is needed. Wtih QUIC it's not needed
-    need_pacing = true;
+  switch(relay_info.proto)
+  {
+    case RelayInfo::Protocol::UDP:
+      [[fallthrough]];
+    case RelayInfo::Protocol::QUIC:
+    client_session =
+      std::make_unique<QuicRClientRawSession>(relay_info, tconfig, logger);
+      break;
+    default:
+      throw QuicRClientException("Unsupported relay protocol");
+      break;
   }
-
-  make_transport(relay_info, std::move(tconfig), logger);
-}
-
-QuicRClient::QuicRClient(std::shared_ptr<ITransport> transport_in)
-  : log_handler(def_log_handler)
-{
-  transport = transport_in;
 }
 
-QuicRClient::~QuicRClient()
+QuicRClient::QuicRClient(std::shared_ptr<qtransport::ITransport> transport_in)
 {
-  removeSubscribeState(true, {},
-                      SubscribeResult::SubscribeStatus::ConnectionClosed);
-  transport.reset(); // wait for transport close
+  client_session =
+    std::make_unique<QuicRClientRawSession>(transport_in);
 }
 
 bool
 QuicRClient::publishIntent(std::shared_ptr<PublisherDelegate> pub_delegate,
                            const quicr::Namespace& quicr_namespace,
-                           const std::string& /* origin_url */,
-                           const std::string& /* auth_token */,
+                           const std::string& origin_url,
+                           const std::string& auth_token,
                            bytes&& payload)
 {
-  if (!pub_delegates.count(quicr_namespace)) {
-    pub_delegates[quicr_namespace] = pub_delegate;
-  }
-
-  messages::PublishIntent intent{ messages::MessageType::PublishIntent,
-                                  messages::create_transaction_id(),
-                                  quicr_namespace,
-                                  std::move(payload),
-                                  transport_stream_id,
-                                  1 };
-
-  messages::MessageBuffer msg{ sizeof(messages::PublishIntent) +
-                               payload.size() };
-  msg << intent;
-
-  auto error =
-    transport->enqueue(transport_context_id, transport_stream_id, msg.get());
-
-  return error == qtransport::TransportError::None;
+  return client_session->publishIntent(
+    pub_delegate, quicr_namespace, origin_url, auth_token, std::move(payload));
```

Comment:
I think std::move might be redundant here, but I don't disapprove of being explicit.

Reply:
Compiler doesn't agree :)